src/pyhf/constraints.py

```diff
@@ -83,10 +83,12 @@ def _precompute(self):
 
     def _dataprojection(self, auxdata):
         tensorlib, _ = get_backend()
-        auxdata = tensorlib.astensor(auxdata)
         normal_data = tensorlib.gather(auxdata, self.normal_data)
         return normal_data
 
+    def has_pdf(self):
```

Comment:
Needs a docstring, but I can add this after we're done with all the benchmark testing so that I don't trigger unnecessarily long builds.